src/transformers/generation/flax_utils.py

```diff
@@ -733,14 +734,20 @@ def beam_search_cond_fn(state):
             not_max_length_yet = state.cur_len < max_length
 
             # 2. can the new beams still improve?
-            best_running_score = state.running_scores[:, -1:] / (max_length**length_penalty)
+            # early_stopping == False -> apply heuristic = always get the best score from `cur_len`
+            # early_stopping == "never" -> compute the best score from max_length or cur_len, depending on the sign of
+            #   length_penalty. Positive length_penalty favors longer sequences, thus we use max_length there.
+            if early_stopping == "never" and length_penalty > 0.0:
+                best_running_score = state.running_scores[:, :1] / (max_length**length_penalty)
+            else:
+                best_running_score = state.running_scores[:, :1] / (state.cur_len**length_penalty)
```

Comment:
Notice the difference in indexing in these lines, `[:, -1:]` -> `[:, :1]`. Why?
1. The running scores are initialized with [0.0, -1e7, -1e7, ...] for each batch, i.e. in descending order. 
2. All other places *were* assuming ascending order of scores.
3. The default case (`early_stopping=False`) has changed. The denominator went from `max_length**length_penalty` to `state.cur_len**length_penalty`. On encoder-decoder models, `state.cur_len=1` on the 1st iteration, meaning that `best_running_score = state.running_scores[:, :1]`
4. Putting these 3 facts together: FLAX's beam search was returning empty outputs, as we were not entering the generation loop 🙃 

There were two possible solutions:
1. Reverse the initialization of the running scores so it is also in ascending order
2. Remove the `jax.flip` operations that were converting `top_k` outputs from descending to ascending order, so that everything stays in descending order

I decided to go with solution 2, since it saves a few flip operations :)